Tests/Kernel/TestPosixFallocate.cpp

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2022, Andreas Kling <kling@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibCore/System.h>
+#include <LibTest/TestCase.h>
+
+TEST_CASE(PosixFallocateBasics)
+{
+    char pattern[] = "/tmp/posix_fallocate.XXXXXX";
+    auto fd = MUST(Core::System::mkstemp(pattern));
+    VERIFY(fd >= 0);
+
+    {
+        auto result = Core::System::posix_fallocate(fd, 0, 1024);
+        EXPECT_EQ(result.is_error(), false);
+
+        auto stat = MUST(Core::System::fstat(fd));
+        EXPECT_EQ(stat.st_size, 1024);
+    }
+
+    {
+        auto result = Core::System::posix_fallocate(-1, 0, 1024);
+        EXPECT_EQ(result.is_error(), true);
+        EXPECT_EQ(result.error().code(), EBADF);
+    }
+
+    {
+        auto result = Core::System::posix_fallocate(fd, 0, -1);
+        EXPECT_EQ(result.is_error(), true);
+        EXPECT_EQ(result.error().code(), EINVAL);
+    }
+
+    {
+        auto result = Core::System::posix_fallocate(fd, 0, 0);
+        EXPECT_EQ(result.is_error(), true);
+        EXPECT_EQ(result.error().code(), EINVAL);
+    }
+
+    {
+        auto result = Core::System::posix_fallocate(fd, -1, 0);
```

Comment:
All 3 of the EINVAL tests have an invalid `length` value (-1, 0 and 0), so they don't properly test if the `offset` parameter is being validated. This specific line could be changed to `posix_fallocate(fd, -1, 1024);` to remedy this

Reply:
`offset` is checked first, so it will currently work as expected, but might break unexpectedly in the future.

Reply:
Sure, let's make sure we're only testing one thing at a time.